uamqp/management_link.py

```diff
@@ -164,18 +179,36 @@ def execute_operation(
             type=None,
             locales=None
     ):
+        message.application_properties["operation"] = operation
+        message.application_properties["type"] = type
+        message.application_properties["locales"] = locales
+        try:
+            # TODO: namedtuple is immutable, which may push us to re-think about the namedtuple approach for Message
+            new_properties = message.properties._replace(message_id=self.next_message_id)
+        except AttributeError:
+            new_properties = Properties(message_id=self.next_message_id)
+        message = message._replace(properties=new_properties)
 
         self._request_link.send_transfer(
             message,
             on_send_complete=self._on_send_complete,
             timeout=timeout
         )
-        self._pending_operations.append(PendingMgmtOperation(message, on_execute_operation_complete))
+        self.next_message_id += 1
+        self._pending_operations.append(PendingManagementOperation(message, on_execute_operation_complete))
 
     def close(self):
         if self.state != ManagementLinkState.IDLE:
             self.state = ManagementLinkState.CLOSING
             self._response_link.detach(close=True)
             self._request_link.detach(close=True)
+            for pending_operation in self._pending_operations:
+                pending_operation.on_execute_operation_complete(
+                    ManagementExecuteOperationResult.INSTANCE_CLOSED,
```

Comment:
"Instance closed"?
To me "instance" in Python refers to an instance of a class... could maybe just go with 'CLOSED', or 'LINK_CLOSED'

Reply:
updated name to LINK_CLOSED